code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -173,6 +173,12 @@ put up a rune with bluespace effects, lots of those runes are fluff or act as a
 			qdel(extract) //vores the slime extract
 			req_extracts--
 			if(req_extracts <= 0)
+				if(extractype == "darkblue")
+					extractype = "dark blue"
+				if(extractype == "darkpurple")
+					extractype = "dark purple"
+				if(extractype == "lightpink")
+					extractype = "light pink"
```

Comment:
this isn't a good way to fix the issue here, just change the names and best define those strings.
also this should be switch not 3 ifs lol

Reply:
yeah a switch would definitly be better just go like 
`					switch(extractype)
						if("darkblue")
							extractype = "dark blue"
						if("darkpurple")
							extractype = "dark purple"
						if("lightpink")
							extractype = "light pink"`
something like this instead of this chain of if's

Reply:
trough then again this issue seems to be purely cause of the fact that there is a space in the string here

Reply:
a better solution would probably be to change the target colors in slime.dm to remove the spaces there of course that would also require to make sure that the things using those slime colors are adapted for that too.

Reply:
Imao the procs actually processing that stuff actually replace the spaces why the hell is this even used then??
```var/sanitizedcolour = replacetext(colour, " ", "")```